packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -669,6 +669,149 @@ void main() {
     expect(renderModel.color, equals(darkTheme.colorScheme.onSurface));
   });
 
+  testWidgets('SnackBar should inherit theme data from its ancestor.', (WidgetTester tester) async {
+    final SliderThemeData sliderTheme = SliderThemeData.fromPrimaryColors(
+      primaryColor: Colors.black,
+      primaryColorDark: Colors.black,
+      primaryColorLight: Colors.black,
+      valueIndicatorTextStyle: const TextStyle(color: Colors.black),
+    );
+
+    final ChipThemeData chipTheme = ChipThemeData.fromDefaults(
+      primaryColor: Colors.black,
+      secondaryColor: Colors.white,
+      labelStyle: const TextStyle(color: Colors.black),
+    );
+
+    const PageTransitionsTheme pageTransitionTheme = PageTransitionsTheme(
+      builders: <TargetPlatform, PageTransitionsBuilder>{
+        TargetPlatform.iOS: CupertinoPageTransitionsBuilder(),
+        TargetPlatform.macOS: CupertinoPageTransitionsBuilder(),
+      },
+    );
+
+    final ThemeData theme = ThemeData.raw(
+      visualDensity: const VisualDensity(),
```

Comment:
If you merge with upstream master, you'll notice that you need three more new theme properties:
```dart
      // ...
      radioTheme: const RadioThemeData(),
      switchTheme: const SwitchThemeData(),
      checkboxTheme: const CheckboxThemeData(),
```